jscomp/runtime/caml_obj.ml

```diff
@@ -132,11 +132,13 @@ let caml_update_dummy (x : Caml_obj_extern.t) (y : Caml_obj_extern.t) : unit =
   done ; 
   let y_tag = Caml_obj_extern.tag y in 
   if y_tag <> 0 then
-    Caml_obj_extern.set_tag x y_tag
+    Caml_obj_extern.set_tag x y_tag *)
 (* Caml_obj_extern.set_length x   (Caml_obj_extern.length y) *)
 (* [set_length] seems redundant here given that it is initialized as an array 
 *)
 
+external caml_update_dummy : Caml_obj_extern.t -> Caml_obj_extern.t -> unit = "Object.assign" [@@bs.val]
+
```

Comment:
caml_update_dummy is used in initializating rec values, how is it related to record encoding?

Reply:
This is used when initializing:
```ocaml
type a = {
  b: b;
}
and b = {
  a: a
};;
```

And it affects `gpr_1716_test.ml`.

Is it used for other things as well, and should there be 2 versions of it?

Reply:
```
type a = {
  b: b;
}
and b = {
  a: a
};;

let rec a = { b } and b = { a };;
```
This seems more subtle. Here the recursive value could be record, but other blocks too, will dig into later